nixos/modules/services/mail/roundcube.nix

```diff
@@ -8,6 +8,11 @@ let
   localDB = cfg.database.host == "localhost";
   user = cfg.database.username;
   phpWithPspell = pkgs.php81.withExtensions ({ enabled, all }: [ all.pspell ] ++ enabled);
+  nginxCommonHeaders = lib.optionalString config.services.nginx.virtualHosts.${cfg.hostName}.forceSSL ''
+    add_header Strict-Transport-Security 'max-age=63072000; includeSubDomains';
```

Comment:
I disagree with this change.

It also seems pretty arbitrary to add this for just the roundcube vhost if `forceSSL` is set, but not for any other vhost with the same configuration (i.e. `virtualHosts."foobar.com"` may also have `forceSSL = true`, but it has no `Strict-Transport-Security` header). This seems really counter-intuitive, unexpected and if someone else will ever need to debug this behavior, it will take a while until this is found out.

At best this should be done for all virtualhosts with `forceSSL = true;` in the nginx module because that is where it belongs[1], however this really seems like hardening that should be done by the system administrator (or abstracted away into a profile by them).

Finally, there's no way to disable that (in whatever corner-case a vhost with `forceSSL = true;` might be needed, but a subdomain may need to work w/o TLS).

[1] I'm __not__ saying that this is a good or bad idea, but if this should be done in NixOS itself, it should be in the nginx module IMHO _unless_ the application requires this header to function properly which doesn't seem to be the case here.

Reply:
> It also seems pretty arbitrary to add this for just the roundcube vhost if `forceSSL` is set, but not for any other vhost with the same configuration (i.e. `virtualHosts."foobar.com"` may also have `forceSSL = true`, but it has no `Strict-Transport-Security` header). This seems really counter-intuitive, unexpected and if someone else will ever need to debug this behavior, it will take a while until this is found out.
> 
> At best this should be done for all virtualhosts with `forceSSL = true;` in the nginx module because that is where it belongs[1], however this really seems like hardening that should be done by the system administrator (or abstracted away into a profile by them).

If you make this option for all hosts, then adding additional headers to ngixn.location will be problematic. With a complex configuration, other headers will be duplicated or other headers will disappear.

> Finally, there's no way to disable that (in whatever corner-case a vhost with `forceSSL = true;` might be needed, but a subdomain may need to work w/o TLS).

I didn’t think about this option, then it’s better to remove `includeSubDomains`.
By default, the `forceSSL` parameter is disabled, but here it is switched to the active state. Then can remove it here.
And add a comment to the `forceSSL` parameter that in some services the HSTS parameter is automatically added.